.github/workflows/publish-e2e-containers.yaml

```
@@ -0,0 +1,87 @@
+
+
+  publish-flightctl-e2e-containers:
+    strategy:
+      matrix:
+        image: 
+          - {name: 'git-server', containerfile: 'test/scripts/agent-images/Containerfile.git-server'}
+          - {name: 'sleep-app', containerfile: 'test/scripts/agent-images/Containerfile.sleep-app'}
+        
+    runs-on: "ubuntu-24.04"
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)